PowerShell.sublime-syntax

```diff
@@ -793,3 +794,17 @@ contexts:
         3: support.variable.drive.powershell
         4: variable.other.readwrite.powershell
         5: punctuation.section.braces.end
+
+  class-method:
+    - match: ^(?:\s*)(?i)(hidden|static)?\s*(\[)((?!\d+|\.)[\p{L}\p{N}.]+)(\])\s*((?:\p{L}|\d|_|-|\.)+)\s*(?=\()
+      scope: meta.function.powershell
+      captures:
+        1: storage.modifier.powershell
+        2: punctuation.section.braces.begin.powershell
+        3: storage.type.powershell
+        4: punctuation.section.braces.end.powershell
+        5: entity.name.function.powershell
+      push:
+        - match: (?=\()
+          pop: true
+        - include: comment-line
```

Comment:
I copied this from the function syntax definition, but what does it do?

Reply:
The whole push doesn't make much sense to me, except if comments required some restrictions to avoid false positives.

Sunction defintion patterns look like a resident of old TextMate syntax.

Today we would do a multi-push to scope function/method argument list, followed by function body.